### PR TITLE
New Python functions `lf.package_directory()` and `lf.source_directory()`

### PR DESCRIPTION
**Pull request by edwardalee**
This PR adds to Python a mechanism to obtain the path of the source directory (the main .lf file) and the package root (where src and bin are).  This mirrors the constants available in the C target `LF_SOURCE_DIRECTORY` and `LF_PACKAGE_DIRECTORY`.

This PR is joint with @vinzbarbuto.

## Summary by CodeRabbit

- **New Features**
  - Introduced new functions for handling directory paths: `py_source_directory` and `py_package_directory`.
- **Enhancements**
  - Improved module functionality related to directory handling with new path-returning features.

### CODE REVIEW

**Review by cmnrd — COMMENTED** [review]
As I pointed out for the corresponding C constants, I think it is a bad practice to expose information about the location of compiled files as static constants to the program. This means that a compiled program cannot be redeployed to a different location.

What is this needed for?

**Comment by cmnrd** [comment]
Why not provide the path via runtime parameters? This could be done via CLI arguments (I guess this is not supported for the python and C targets) or by reading an environment variable.

**Comment by lhstrh** [comment]
@cmnrd:I think the conclusion we reached during our last call was that we can admit this as a temporary solution?

File: python/lib/pythontarget.c [hunk 1]
```diff
@@ -304,6 +312,8 @@
                                                         {"tag", py_lf_tag, METH_NOARGS, NULL},
                                                         {"tag_compare", py_tag_compare, METH_VARARGS, NULL},
                                                         {"request_stop", py_request_stop, METH_NOARGS, "Request stop"},
+                                                        {"source_directory", py_source_directory, METH_NOARGS, "Source directory path for .lf file"},
+                                                        {"package_directory", py_package_directory, METH_NOARGS, "Root package directory path"},
```
[the remark this answers is not in the document]
Comment (edwardalee) [reply python/lib/pythontarget.c]: This complaint from coderabbitai is incorrect.  These functions _must_ have two argument.

**Review by cmnrd — APPROVED** [review]
Looks good!

**Pull request merged**